engine/graphic/src/Texture.cpp

```diff
@@ -0,0 +1,86 @@
+#include "Texture.h"
+Texture::Texture(char* FileName)
+{
+	m_fileName = FileName;
+	imageId = 0;
+	mGlTextureObject = 0;
+}
+
+bool Texture::Load()
+{
+	ilGenImages(1, &imageId);
+	// ������ ����������� ������� 
+	ilBindImage(imageId);
+
+	// ���� �������� �������
+	if (ilLoadImage(m_fileName))
+	{
+		// ���� �������� ������ ������� 
+		// ��������� ������� ����������� 
+		int width = ilGetInteger(IL_IMAGE_WIDTH);
+		int height = ilGetInteger(IL_IMAGE_HEIGHT);
+
+		// ���������� ����� ��� �� ������� 
+		int bitspp = ilGetInteger(IL_IMAGE_BITS_PER_PIXEL);
+
+		switch (bitspp)// � ����������� �� ����������� ���������� 
+		{
+			// ������� �������� ��������� ����� GL_RGB ��� GL_RGBA 
+		case 24:
+			mGlTextureObject = MakeGlTexture(GL_RGB, ilGetData(), width, height);
+			break;
+		case 32:
+			mGlTextureObject = MakeGlTexture(GL_RGBA, ilGetData(), width, height);
+			break;
+		}
+
+		// ������� ������ 
+		ilBindImage(0);
```

Comment:
It's not GL function, though. So I really can't understand how I can bind atlases to different buffers